test/functional/action-params.spec.ts

```diff
@@ -72,6 +78,46 @@ describe("action parameters", () => {
         // reset metadata args storage
         defaultMetadataArgsStorage().reset();
 
+
+        @Middleware()
+        class SessionMiddleware implements MiddlewareInterface {
```

Comment:
I'm not sure, have you tried?
In other test suites the middlewares are defined with the test code so it can stay here were it is.
The `/test/fakes/` folder is used only for testing loading middlewares and error handlers from the directories.